runtime/compiler/runtime/MetaData.cpp

```diff
@@ -158,6 +159,22 @@ createByteCodeInfoRange(
    location += sizeof(int32_t);
    }
 
+/*
+ * use special catch type value to indicate to the VM the handler is synthetic handler so that OSR can be handled correctly
+ */
+static uint32_t
+useSpecialCatchTypeIfNecessary(TR_ExceptionTableEntry *e, bool fourByteOffsets, TR::Compilation *comp)
+   {
+   if (comp->getOption(TR_EnableOSR))
+      {
+      if (fourByteOffsets && e->_catchType == J9_CATCHTYPE_VALUE_FOR_THYNTHETIC_HANDLER_4BYTES ||
```

Comment:
Some brackets would be good here (and perhaps essential - && and || likely have exactly the same precedence properties).

Also synthetic is misspelled.

Reply:
@cathyzhyi please address above and I'll restart the build.

Reply:
Thanks for the comments. changed as suggested.